circleci: failed tests (15796): TestTxnRestart

The following test appears to have failed:

[#15796](https://circleci.com/gh/cockroachdb/cockroach/15796):

```
I160404 01:53:39.060727 storage/replica_command.go:539  auto-gc'ed ce4178c3 (4 intents)
I160404 01:53:39.062390 storage/replica_command.go:74  test injecting error: read at time 0.000000000,0 encountered previous write with future timestamp 0.000000000,0 within uncertainty interval
W160404 01:53:39.062768 kv/dist_sender.go:684  failed to invoke BeginTransaction [/Table/51/1/"g"/0,/Min), ConditionalPut [/Table/51/1/"g"/0,/Min), ConditionalPut [/Table/51/1/"g"/2/1,/Min), EndTransaction [/Table/51/1/"g"/0,/Min): read at time 0.000000000,0 encountered previous write with future timestamp 0.000000000,0 within uncertainty interval
I160404 01:53:39.062831 server/updates.go:147  No previous updates check time.
I160404 01:53:39.062981 client/db.go:476  failed batch: read at time 0.000000000,0 encountered previous write with future timestamp 0.000000000,0 within uncertainty interval
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x7a32d9]

goroutine 50464 [running]:
panic(0x167cc80, 0xc8200100e0)
	/usr/local/go/src/runtime/panic.go:464 +0x3e6
github.com/cockroachdb/cockroach/sql_test.injectErrors(0x7fa4d0884808, 0xc8204c19e0, 0x144205157c87db73, 0x0, 0x100000001, 0x1, 0x1, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/sql/txn_restart_test.go:113 +0x2b9
github.com/cockroachdb/cockroach/sql_test.TestTxnRestart.func1(0x7fa4d08ce1d8, 0xc8205a4570, 0xc82060fb18, 0x8, 0x0, 0x1, 0x7fa4d0884808, 0xc8204c19e0, 0x144205157c87db73, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/sql/txn_restart_test.go:241 +0x5d
github.com/cockroachdb/cockroach/sql_test.(*CommandFilters).runFiltersInternal(0xc820012140, 0x7fa4d08ce1d8, 0xc8205a4570, 0xc82060fb18, 0x8, 0x0, 0x1, 0x7fa4d0884808, 0xc8204c19e0, 0x144205157c87db73, ...)
	/go/src/github.com/cockroachdb/cockroach/sql/main_test.go:78 +0xe2
github.com/cockroachdb/cockroach/sql_test.(*CommandFilters).(github.com/cockroachdb/cockroach/sql_test.runFiltersInternal)-fm(0x7fa4d08ce1d8, 0xc8205a4570, 0xc82060fb18, 0x8, 0x0, 0x1, 0x7fa4d0884808, 0xc8204c19e0, 0x144205157c87db73, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/sql/main_test.go:108 +0x3a
github.com/cockroachdb/cockroach/testutils/storageutils.(*ReplayProtectionFilterWrapper).run(0xc820a45980, 0x7fa4d08ce1d8, 0xc8205a4570, 0xc82060fb18, 0x8, 0x0, 0x1, 0x7fa4d0884808, 0xc8204c19e0, 0x144205157c87db73, ...)
--
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:1869 +0x35a
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820a68000, 0xc8213db8c0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:139 +0x52
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:140 +0x62
FAIL	github.com/cockroachdb/cockroach/sql	42.067s
=== RUN   TestEval
--- PASS: TestEval (0.01s)
=== RUN   TestEvalError
--- PASS: TestEvalError (0.00s)
=== RUN   TestEvalComparisonExprCaching
--- PASS: TestEvalComparisonExprCaching (0.00s)
=== RUN   TestSimilarEscape
--- PASS: TestSimilarEscape (0.00s)
=== RUN   TestQualifiedNameString
--- PASS: TestQualifiedNameString (0.00s)

```
Please assign, take a look and update the issue accordingly.

--- a/sql/txn_restart_test.go
+++ b/sql/txn_restart_test.go
@@ -110,7 +110,6 @@ func injectErrors(
 
 	switch req := req.(type) {
 	case *roachpb.ConditionalPutRequest:
-		txnID := *hdr.Txn.TxnMeta.ID
 		for key, count := range magicVals.restartCounts {
 			checkCorrectTxn(string(req.Value.RawBytes), magicVals, hdr.Txn)
 			if count > 0 && bytes.Contains(req.Value.RawBytes, []byte(key)) {
@@ -136,6 +135,7 @@ func injectErrors(
 		// keep track of the txn id so we can fail it later on.
 		for key, count := range magicVals.endTxnRestartCounts {
 			if count > 0 && bytes.Contains(req.Value.RawBytes, []byte(key)) {
+				txnID := *hdr.Txn.TxnMeta.ID
 				if _, found := magicVals.txnsToFail[txnID]; found {
 					continue
 				}

